rllib/algorithms/appo/appo.py

```diff
@@ -158,15 +172,52 @@ def training(
             self.lambda_ = lambda_
         if clip_param is not NotProvided:
             self.clip_param = clip_param
+            self._learner_hps.clip_param = clip_param
         if use_kl_loss is not NotProvided:
             self.use_kl_loss = use_kl_loss
         if kl_coeff is not NotProvided:
             self.kl_coeff = kl_coeff
+            self._learner_hps.kl_coeff = kl_coeff
         if kl_target is not NotProvided:
             self.kl_target = kl_target
+            self._learner_hps.kl_target = kl_target
+        if tau is not NotProvided:
+            self.tau = tau
+            self._learner_hps.tau = tau
+        if target_update_frequency is not NotProvided:
+            self.target_update_frequency = target_update_frequency
 
         return self
 
+    @override(AlgorithmConfig)
+    def get_default_learner_class(self):
+        if self.framework_str == "tf2":
+            from ray.rllib.algorithms.appo.tf.appo_tf_learner import APPOTfLearner
+
+            return APPOTfLearner
+        else:
+            raise ValueError(f"The framework {self.framework_str} is not supported.")
+
+    @override(AlgorithmConfig)
+    def get_default_rl_module_spec(self) -> SingleAgentRLModuleSpec:
+        if self.framework_str == "tf2":
+            from ray.rllib.algorithms.ppo.ppo_catalog import PPOCatalog
+            from ray.rllib.algorithms.appo.tf.appo_tf_rl_module import APPOTfRLModule
+
+            return SingleAgentRLModuleSpec(
+                module_class=APPOTfRLModule, catalog_class=PPOCatalog
+            )
+        else:
+            raise ValueError(f"The framework {self.framework_str} is not supported.")
+
+    @override(ImpalaConfig)
+    def validate(self) -> None:
+        super().validate()
+        self._learner_hps.tau = self.tau
```

Comment:
are you asking why we have to set learner hps inside of self.validate? It is because otherwise they will not be set when users are creating an algorithm directly from a config dict.

Reply:
I mean this is not he correct place to do it. It should be done in the base class. You don't expect this to be done for every algorithm, right? Here is how you should do it: 
In algorithm_config in the method `update_from_dict`, in the last else statement here is what is happening right now:

```
if not hasattr(self, key) and log_once(
    "unknown_property_in_algo_config"
):
    logger.warning(
        f"Cannot create {type(self).__name__} from given "
        f"`config_dict`! Property {key} not supported."
    )
setattr(self, key, value)
```

This is roughly what should be happening:
```
if not hasattr(self, key) and log_once(
    "unknown_property_in_algo_config"
):
    logger.warning(
        f"Cannot create {type(self).__name__} from given "
        f"`config_dict`! Property {key} not supported."
    )
setattr(self, key, value)

if hasattr(self._learner_hps, key):
      setattr(self._learner_hps, key, value)

```

Reply:
See my comment above on an HP "compilation" method to avoid duplicate values for the same thing in different places.

Reply:
going to add a separate function in algorithm config.py for building learner hp objects